File: 알고리즘/1/비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(len(arr1)):
      
      a1 = bin(arr1[i]).replace('ob', '')
      b1 = bin(arr2[i]).replace('ob', '')

      a1 = a1[-n:-1]
      b1 = b1[-n:-1]
      
      s = ''
      for j in range(n):
        if a1[j]=='0' and b1[j]=='0': 
          s+=' '
        else: s+='#'
      answer.append(s)

    return answer

solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28])

File: 알고리즘/1/l1/비밀지도.cpp
#include <bits/stdc++.h>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    
    for (int i=0; i<n; i++){
      auto a = bitset<16> (arr1[i]|arr2[i]).to_string();
      auto a1 = a.substr(distance(a.begin(), a.end()-n), n);
      replace(a1.begin(), a1.end(), '1', '#');
      replace(a1.begin(), a1.end(), '0', ' ');
      answer.push_back(a1);
    }
    return answer;
}